### DISCUSSION TITLE
Mengekstrak Nilai Sebuah Kolom

### DISCUSSION THREAD

**Discussion by ToniAS21** [teamalgoritma/community#236] [Q&A] [answered]
Permisi mau nanya apakah ada cara agar kolom `Longname` dari data dow_companies dapat diekstrak ke data dow_stock ? Jadi fokusnya ingin data dow_stock memiliki kolom baru namanya `Longname` yg mana nilai kolom ini disesuaikan dengan `Symbol`. Outputnya pada data dow_stock, semisal dibagian kolom `Symbol` "AAPL" berarti dibagian kolom `Longname` "Apple.Inc."
![IMG-20220802-WA0000](https://user-images.githubusercontent.com/83503249/182271340-812af998-30e1-49e7-b255-e73c814eee2c.jpg)

Sekian terima kasih

**Comment by amaliasf** [comment, marked as answer]
Halo Pak @ToniAS21 ,
Untuk menggabungkan 2 dataframe tersebut bapak bisa menggunakan function `merge()` pada package `tidyverse`. Kemudian dikombinasikan dengan subsetting karna hanya kolom `Longname` yang ingin ditambahkan.

Berikut contoh code yang bisa diterapkan:
`merge(x = dow_stock, y = dow_companies[ , c("Symbol", "Longname")], by = "Symbol", all.x=TRUE)`

Referensi: 
- [Merge with Selected Columns in R](https://stackoverflow.com/questions/24191497/left-join-only-selected-columns-in-r-with-the-merge-function)
- [Joining of DataFrames in R]( https://www.geeksforgeeks.org/joining-of-dataframes-in-r-programming/#:~:text=In%20R%20we%20use%20merge,similarly%20like%20join%20in%20DBMS.)

**Reply by ToniAS21** [reply]
Baik terima kasih banyak Bu @amaliasf . Sudah bisa.
